### PR TITLE
fzf-obc: init at 1.3.0

### PR DESCRIPTION
**Pull request by loicreynier**
###### Description of changes

Add [fzf-obc](https://github.com/rockandska/fzf-obc), a Bash completion script adding fzf mechanisms to all completion functions.

Because of the unconventional structure for a shell plugin, I opted for a package `opt/{bin,lib,plugin}` package structure. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/bash/fzf-obc/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    install -m644 bin/* $out/opt/${pname}/bin
+    install -m644 lib/${pname}/* $out/opt/${pname}/lib/${pname}
+    install -m644 plugins/kill/* $out/opt/${pname}/plugins/kill
+    install -m644 plugins/gradle/* $out/opt/${pname}/plugins/gradle
+  '';
+
+  meta = with lib; {
+    homepage = "https://fzf-obc.readthedocs.io";
+    description = "Completion script adding fzf over all know bash completion functions";
+    license = "unknown";
```
Comment (SuperSandro2000) [inline pkgs/shells/bash/fzf-obc/default.nix:29]:
```suggestion
    license = licenses.unfree;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Why are we copying this into opt? Can't we copy this under $out/share or something more FHS?

**Comment by loicreynier** [comment]
> Why are we copying this into opt? Can't we copy this under $out/share or something more FHS?

I opted for `opt` because of the unconventional structure but `share` is fine to me too.

**Comment by SuperSandro2000** [comment]
nixpkgs rarely uses opt. Sometimes for propiatery programs or web apps but I think using something else here would be nicer. Can you change that?

File: pkgs/shells/bash/fzf-obc/default.nix (hunk 1 above)
Comment (x10an14) [reply pkgs/shells/bash/fzf-obc/default.nix]: Hey @SuperSandro2000, where's the guideline saying that for nixpkgs' packages, a missing license implies defaulting to unfree?

**Pull request merged**